### PR TITLE
reference/performance: elaborate more on optimizer hints

### PR DESCRIPTION
**Pull request by eurekaka**
### What is changed, added or deleted? (Required)

Fix https://github.com/pingcap/docs-cn/issues/2481

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

N/A

### CODE REVIEW

**Comment by yikeke** [comment]
PTAL @SunRunAway @lilinghai

File: reference/performance/optimizer-hints.md [hunk 1]
```diff
@@ -13,58 +13,66 @@ TiDB 支持 Optimizer Hints 语法，它基于 MySQL 5.7 中介绍的类似 comm
...
 
-## 优化器相关 Hint 语法
-
-用于控制优化器行为的 Hint 跟在语句中**任意** `SELECT`、`UPDATE` 或 `DELETE` 关键字的后面。Hint 的生效范围以及 Hint 中使用的表的生效范围可通过 [Query Block](#query-block) 来指定，若不显式地指定 Query Block， Hint 的默认生效范围为当前 Query Block。
+> **注意：**
+>
+> 如果注释不是跟在指定的关键字后，会被当作是普通的 MySQL comment，注释不会生效，且不会上报 warning。
 
-### Query Block
+TiDB 目前支持的 Optimizer Hints 根据生效范围的不同可以划分为两类：第一类是在查询块范围生效的 Hint，例如 [`/*+ HASH_AGG() */`](#hash_agg)；第二类是在整个查询范围生效的 Hint，例如 [`/*+ MEMORY_QUOTA(1024 MB)*/`](#memory_quota)。
```
Comment (ran-huang) [inline reference/performance/optimizer-hints.md:30]:
```suggestion
TiDB 目前支持的 Optimizer Hints 根据生效范围的不同可以划分为两类：第一类是在查询块范围生效的 Hint，例如 [`/*+ HASH_AGG() */`](#hash_agg)；第二类是在整个查询范围生效的 Hint，例如 [`/*+ MEMORY_QUOTA(1024 MB)*/`](#memory_quotan)。
```

**Comment by ran-huang** [comment]
@eurekaka @lilinghai @SunRunAway PTAL~

File: reference/performance/optimizer-hints.md [hunk 2]
```diff
@@ -13,58 +13,66 @@ TiDB 支持 Optimizer Hints 语法，它基于 MySQL 5.7 中介绍的类似 comm
...
+Optimizer Hints 通过 `/*+ ... */` 注释的形式跟在 `SELECT`、`UPDATE` 或 `DELETE` 关键字的后面(`INSERT` 关键字后不支持 Optimizer Hints)，常见形式如 `/*+ HINT_NAME([t1_name [, t2_name] ...]) */`。Hint 名称不区分大小写，多个不同的 Hint 之间需用逗号隔开。例如：
 
 {{< copyable "sql" >}}
 
 ```sql
 select /*+ USE_INDEX(t1, idx1), HASH_AGG(), HASH_JOIN(t1) */ count(*) from t t1, t t2 where t1.a = t2.b;
 ```
 
-TiDB 目前支持两类 Hint，具体用法上有一些差别。第一类 Hint 用于控制优化器的行为，例如 [`/*+ HASH_AGG() */`](#hash_agg)。第二类 Hint 用于对单条查询设置一些运行参数，例如 [`/*+ MEMORY_QUOTA(1024 MB)*/`](#memory_quotan)。
+Optimizer Hints 可以和 `Explain` / `Explain Analyze` 组合使用，通过这两个命令查看验证 Optimizer Hints 是否按照预期对查询产生了影响。如果 Optimizer Hints 包含语法错误，或者不适用于当前语句，查询会按照没有 Optimizer Hints 的情况执行，不会对 Optimizer Hints 部分报错，而是会记录 Warning，用户可以在查询结束后通过 `Show Warnings` 命令查看具体信息。
```
[the remark this answers is not in the document]
Comment (eurekaka) [reply reference/performance/optimizer-hints.md:24]: No

**Pull request merged**